### DISCUSSION TITLE
No TelegraphChat defined for this request

### DISCUSSION THREAD

**Discussion by Nurbolaaaat** [def-studio/telegraph#438] [Q&A] [answered]
Hi ! 

The bot works only with one user. When I write to the bot from a second user, the bot starts to lag and does not respond to anyone.

![image](https://github.com/defstudio/telegraph/assets/138411350/fffbc5bc-c57f-4f2d-a30c-540b7e236a9e)
![image](https://github.com/defstudio/telegraph/assets/138411350/f4b48fbf-841c-499f-ad1d-a2c16664873a)
![image](https://github.com/defstudio/telegraph/assets/138411350/fac4fc1c-da30-40bd-9440-89e1c383f758)

**Comment by fabio-ivona** [comment, marked as answer]
Hi!

could you put some logs here: `vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php`  to confirm that the method `setupChat()` is called on both requests?

**Reply by zorlovskiy** [reply]
Здравствуйте, возникла такая же проблема. Вот часть того, что записано в laravel.log

[2023-12-20 15:41:12] local.DEBUG: сообщение телеграфного веб-перехватчика {"text":"/buttons"}
[2023-12-20 15:41:12] local.ERROR: для этого запроса не определен TelegraphChat {"исключение":"[объект] (DefStudio\Telegraph\Exceptions\TelegraphException(код: 0): для этого не определен TelegraphChat запрос в /application/vendor/defstudio/telegraph/src/Exceptions/TelegraphException.php:16)
[трассировки стека]
#0 /application/vendor/defstudio/telegraph/src/Concerns/HasBotsAndChats.php(116): DefStudio\Telegraph\Exceptions\TelegraphException::missingChat()
https://github.com/defstudio/telegraph/pull/1/application/vendor/defstudio/telegraph/src/Concerns/HasBotsAndChats.php(121): DefStudio\Telegraph\Telegraph->getChat()
https://github.com/defstudio/telegraph/pull/2/application/vendor/defstudio/telegraph/src/Concerns/ComposesMessages.php(28): DefStudio\Telegraph\Telegraph->getChatId()
https://github.com/defstudio/telegraph/pull/3/application/vendor/defstudio/telegraph/src/Concerns/ComposesMessages.php(36): DefStudio\Telegraph\Telegraph->setMessageText()
https://github.com/defstudio/telegraph/pull/4/application/vendor/defstudio/telegraph/src/Concerns/ComposesMessages.php(19): DefStudio\Telegraph\Telegraph->html()
https://github.com/defstudio/telegraph/pull/5/application/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(355): DefStudio\Telegraph\Telegraph->message()
https://github.com/defstudio/telegraph/pull/6/application/app/CommandHandler/CommandHandler.php(36): Illuminate\Support\Facades\Facade::__callStatic()
https://github.com/defstudio/telegraph/issues/7/application/vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php(81): App\CommandHandler\CommandHandler->buttons()
https://github.com/defstudio/telegraph/pull/8/application/vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php(108): DefStudio\Telegraph\Handlers\WebhookHandler->handleCommand()
https://github.com/defstudio/telegraph/pull/9/application/vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php(228): DefStudio\Telegraph\Handlers\WebhookHandler->handleMessage()
https://github.com/defstudio/telegraph/pull/10/application/vendor/defstudio/telegraph/src/Controllers/WebhookController.php(28): DefStudio\Telegraph\Handlers\WebhookHandler->handle()

**Reply by AmnesiaZero** [reply]
Same error here. Checked database,data about new chat is stored,but bot stops working with same error. When i delete new chat from DB, bot starts working

**Reply by fabio-ivona** [reply]
@AmnesiaZero 

could you put some logs here: vendor/defstudio/telegraph/src/Handlers/WebhookHandler.php to confirm that the method setupChat() is called on both requests?

**Comment by AmnesiaZero** [comment]
@Nurbolaaaat yeah,it's logged

**Reply by AmnesiaZero** [reply]
I fixed my error. I changed all $telegrpah->message() to $telegraph->chat()->message and it's working

**Reply by fabio-ivona** [reply]
yeah, telegraph requires chat() when replying
